lib/blocktree/blocktree.go

```diff
@@ -368,5 +394,9 @@ func (bt *BlockTree) DeepCopy() *BlockTree {
 		}
 	}
 
+	for hash := range bt.nodeCache {
+		btCopy.nodeCache[hash] = btCopy.getNode(hash)
```

Comment:
this still references the node `bt`, you will need to allocate a new `*node` and copy the fields over to create a deep copy

Reply:
Aren't the nodes in `btCopy` already a deepCopy of each node in `bt`??

Reply:
nevermind, I missed that. this is fine then. however since `getNode` already places it in the cache, you can change this line to `_ = btCopy.getNode(hash)`

Reply:
Yep, i assigned the value just to make more explicit the intention of the code, but i can remove it if you want.